cranelift/codegen/src/isa/riscv64/inst.isle

```diff
@@ -1904,7 +1904,7 @@
 (rule (normalize_value $I16 r)
       (value_reg (alu_rrr (AluOPRRR.And) r (imm $I16 65535))))
 (rule (normalize_value $I32 r)
-      (value_reg (alu_rr_imm12 (AluOPRRI.Andi) r (imm12_const -1))))
+      (value_reg (alu_rrr (AluOPRRR.And) r (imm $I32 0xffffffff))))
```

Comment:
I'm not too familiar with RISCV, is there a better way of masking off the top 32bits? This ends up generating a lot of code, which is unfortunate.

Reply:
For reference, this generates the following assembly:
```
;   auipc t2,0
;   ld t2,12(t2)
;   j 12
;   .8byte 0xffffffff
;   and a4,a0,t2
```

The only other solution I can think of right now would be to shift left 32 and then shift right 32, which would only be 2 instructions.

Reply:
Playing around with clang's output on godbolt shows that it does this conversion with `addiw rd, rs, 0` (or the `sext.w` pseudo instruction). That will perform a 32-bit add of `0` to `rs`, and then sign-extend the value to 64-bits. As we do appear to be using `srliw` for a 32-bit right shift, I think this should work?

Reply:
For `normalize_value` I think that is not the correct answer since it always guarantees that the top 32 bits will be clear, which may not always be the case for `sext.w`. However I think we can use that for `bmask`/`brz`/`brnz` and we get back to a 1 instruction solution which is nice!

Reply:
`normalize_value` might be overly conservative as it is now. Switching to `addiw rd, rs, 0` for the 32-bit case didn't produce any test failures for me locally, and I suspect that we don't even need that instruction.

Reply:
I've updated it to use `addiw` and  changed the meaning of `normalize_value` so that it fits the new implementation. Thanks!

> I suspect that we don't even need that instruction.
That would be even better, if you have any ideas about this let me know!